R/zchunk_L2233.electricity_water.R

```diff
@@ -84,406 +111,898 @@ module_water_L2233.electricity_water_DISABLED <- function(command, ...) {
+
+    L2233.supplysector_info %>%
+      left_join_error_no_match(supply_sub_elec_mapping, by = "supplysector") %>%
+      mutate(year.fillout = first(BASE_YEARS)) %>%
+      mutate(share.weight = 1) %>%
+      select(one_of(LEVEL2_DATA_NAMES$SubsectorShrwtFllt)) ->
+      L2233.SubsectorShrwtFllt_elec_cool # --OUTPUT--
+
+    L2233.SubsectorShrwtFllt_elec_cool %>%
+      mutate(logit.year.fillout = first(BASE_YEARS)) %>%
+      mutate(logit.exponent = COOLING_SYSTEM_LOGIT) %>%
+      select(one_of(LEVEL2_DATA_NAMES$SubsectorLogit)) %>%
+      mutate(logit.type = NA) -> L2233.SubsectorLogit_elec_cool # --OUTPUT--
+
+
+    ## PART 3: GLOBAL TECHNOLOGY INFORMATION RE-ASSIGNED TO NEW STRUCTURE
+
+    # Notes: (1) All global technology information from the prior electricity sector...
+    #            ...is carried over to the new locations;
+    #        (2) We use the string "sector.name" to identify tables with global technologies...
+    #            ... since not all have "GlobalTech" in their name
+
+    # Filter the L223 files list for those with "sector.name" as a column header
+    sctrFltr <- sapply(L223_data, function(x) "sector.name" %in% attributes(x)$names)
+    L2233.Elec_tables_globaltech <- L223_data[sctrFltr]
+
+    # If the table has cost information, then we keep it in the passthrough technology...
+    # ... because we don't want the whole-plant costs to be bundled with the cooling...
+    # ... system costs. Keeping these together would mean that cooling system decisions...
+    # ... would be made on < 1% cost differences, which would require very high logit...
+    # ... to get any behavior at all.
+    costFltr <- sapply(L2233.Elec_tables_globaltech, function(x)
+      grepl("Capital", attributes(x)$title) |
+        grepl("OM", attributes(x)$title))
+    L2233.Elec_tables_globaltech_cost <- L2233.Elec_tables_globaltech[costFltr]
+    # ^^ filters for tables with capital, fixed O&M and variable O&M
+
+    # We reset the intermittent technology to standard technologies that remain in ...
+    # ... the electricity sector. For an intermittent technology to be moved to a new ...
+    # ... sector, the new passthur tech in the electricity sector is no longer an ...
+    # ... intermittent technology (this would double count).Thus, for all L2233.Elec_tables_globaltech_cost, ...
+    # ... we change "intermittent.technology" to "technology".
+
+    elec_tech_water_map %>%
+      select(-plant_type, - cooling_system, - water_type,
+             -sector, -fuel, -technology, -minicam.energy.input) %>%
+      rename(sector.name = from.supplysector,
+             subsector.name = from.subsector,
+             technology = from.technology) %>%  unique -> elec_tech_water_map_
+    # ^^ reduce and rename elec_tech_water_map for joining to all tables in L2233.Elec_tables_globaltech_cost
+
+    resetTech <- function(x){
+      names(x)[names(x) == "intermittent.technology"] <- "technology"
```

Comment:
argh, this one slipped through the net when I was going through all the corrections... looking at though, I think I used this approach because `rename` will return an error in instances where the column header isn't actually in the tibble (this code is applied across several tibbles, some of which won't need any column name changes). If it's imperative to use `rename` let me know and I'll come up with something.